packages/shared/src/features/briefing/components/BriefPayForGenerateCard.tsx

```diff
@@ -0,0 +1,172 @@
+import { CoreIcon } from '../../../components/icons';
+import { BriefingType } from '../../../graphql/posts';
+import { ActionType } from '../../../graphql/actions';
+import { useGenerateBrief } from '../hooks/useGenerateBrief';
+import { ClickableCard } from '../../../components/cards/common/Card';
+import { Radio, type RadioProps } from '../../../components/fields/Radio';
+import {
+  Button,
+  ButtonSize,
+  ButtonVariant,
+} from '../../../components/buttons/Button';
+import { BuyCoresModal } from '../../../components/modals/award/BuyCoresModal';
+import { useToggle } from '../../../hooks/useToggle';
+import { Origin } from '../../../lib/log';
+import type { Product } from '../../../graphql/njord';
+import { useFeature } from '../../../components/GrowthBookProvider';
+import { briefGeneratePricing } from '../../../lib/featureManagement';
+
+const OPTIONS = [
+  { value: BriefingType.Daily, label: 'Daily - last 24 hours' },
+  { value: BriefingType.Weekly, label: 'Weekly - last 7 days' },
+] as const satisfies RadioProps['options'];
+
+function createCoresProduct(id: string, name: string, value: number): Product {
+  // Minimal shape required by BuyCoresModal; value is what matters.
+  return { id, name, value } as unknown as Product;
+}
+
+function BuyCores({
+  isOpen,
+  onClose,
+  price,
+  type,
+  onPurchased,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  price: number;
+  type: BriefingType;
+  onPurchased: () => void;
+}) {
+  const product = useMemo(
+    () => createCoresProduct(`brief-${type}`, 'Briefing', price),
+    [price, type],
+  );
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <BuyCoresModal
+      isOpen={isOpen}
+      onRequestClose={onClose}
+      onCompletion={() => {
+        onClose();
+        onPurchased();
+      }}
+      product={product}
+      origin={Origin.BriefPage}
+    />
+  );
+}
+
+export const BriefPayForGenerateCard = () => {
+  const router = useRouter();
+  const { user } = useAuthContext();
+  const { displayToast } = useToastNotification();
+  const { checkHasCompleted, isActionsFetched, completeAction } = useActions();
+
+  const [selected, setSelected] = React.useState<BriefingType>(
+    BriefingType.Daily,
+  );
+  const [isBuyOpen, setBuyOpen] = useToggle(false);
+
+  const prices = useFeature(briefGeneratePricing);
+  const price = prices[selected];
+
+  const isFirstBrief =
+    isActionsFetched && !checkHasCompleted(ActionType.GeneratedBrief);
+  const amount = user?.balance?.amount ?? 0;
+
+  const { isGenerating, generateBrief } = useGenerateBrief({
+    onSuccess: async () => {
+      displayToast('Your Presidential Briefing is being generated ✅');
+      await Promise.all([
+        completeAction(ActionType.GeneratedBrief),
+        router.push('/briefing'),
+      ]);
+    },
+    onError: () => {
+      displayToast(
+        'There was an error generating your Presidential Briefing. Please try again later.',
+      );
```

Comment:
Since users now pay with Cores we should indicate brief was paid with Cores and log it on the API side for handling and potentially retrying, this error message is ok here but might be a good point to raise with product and see how they want to handle it.